### DISCUSSION TITLE
Availability of the `uap10-xx` target framework in 1.0

### DISCUSSION THREAD

**Discussion by jeromelaban** [microsoft/WindowsAppSDK#1655] [Q&A] [answered]
As the UWP target has been officially announced as not being supported by the WinAppSDK, what will be the fate of the `uap10.0` projections in the `Microsoft.WindowsAppSDK.*` packages? 

Are they still going to be included, and for how long?

![image](https://user-images.githubusercontent.com/5839577/139088416-61649125-d4ba-4d27-8249-dea329ec9a7f.png)

**Comment by riverar** [comment]
+1. I currently use these to generate Rust bindings. So if I need to consume something else, I'll need that new pointer.

**Reply by jeromelaban** [reply]
The net5-desktop bindings are significantly different in many areas.

For instance, there are changes like these to take into account:
![image](https://user-images.githubusercontent.com/5839577/139123365-707aa88f-756b-4b8f-9d99-ad464e59c6b5.png)

**Reply by andrewleader** [reply]
Hey Rafael and Jerome, I'm starting a discussion internally about this. To collect more info, if we did remove the uap10.0 folders, **would switching to use the net5.0-desktop folders be a blocker for your projects**? And if so, why? Can you explain how you're using them?

Are you essentially using the uap10.0 winmds to generate Rust bindings that then operate on the desktop-based DLLs?

**Reply by jeromelaban** [reply]
For Uno Platform, it's not a technically blocker but rather more work as the desktop APIs are quite different from the uap ones. For instance, there are lots of public interop methods that are surfaced that would benefit from being flagged as non-browsable or that could at least be discriminated from the "useful" APIs.

For the example above, are the `_x` members supposed to be public visibility?

**Reply by andrewleader** [reply]
That makes sense, thanks Jerome. I assume the `_x` members aren't supposed to be publicly visible, but I can check about that.

I've also confirmed internally with our team that these uap winmds are used for native projections (C++/Rust).

If we made these same previous winmds available in a `native` TFM folder, would that still work for you? We're still very early in investigating and just a potential idea.

**Reply by jeromelaban** [reply]
Thanks for the updates! I can see that Preview 3 has also aligned the Input APIs in the uap projections, that'll help a lot with generating bindings.

The location of the WinMDs are not necessarily a problem, as long as they are available.

**Reply by manodasanW** [reply]
Jerome, the `_x` members here should have been marked internal but fixing that now is technically a breaking change for us given it is already part of the public API surface.  I see you mention flagging them as non-browsable.  If we were to add the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute to both members, would that help you?  I assume your concern is seeing them with some tooling where you can do filtering or is it with seeing those members in VS IntelliSense?

**Reply by riverar** [reply]
@andrewleader There is no metadata present in the .net5.0 folders you described. Those are assemblies that provide .NET language specific projections (e.g. C#).

> Are you essentially using the uap10.0 winmds to generate Rust bindings that then operate on the desktop-based DLLs?

All **non-.NET** languages (C++, Rust, Dart, Zig, etc.) will need neutral metadata to create language projections.

> If we made these same previous winmds available in a `native` TFM folder, would that still work for you? We're still very early in investigating and just a potential idea.

As long as they continue to exist, feel free to move them wherever you want.

**Reply by riverar** [reply]
@manodasanW Preview to 1.0 GA can break. Now is your chance before it's too late (until 2.0).

**Reply by jeromelaban** [reply]
I'd say that from a developer experience point of view, having those fields is particularly misleading. Hiding them from intellisense is one step, but removing them completely is certainly best, as @riverar said. Hiding the other interop members will also provide a better experience.

**Reply by jeromelaban** [reply]
Also, this is the WinAppSDK project, but will the WinUI projections do the same move to the native folder ?

**Reply by andrewleader** [reply]
Thanks everyone! @jeromelaban if we move the projections, since we now have one single NuGet, we'd move all of them (including WinUI) to the same folder (whether that's a `native` folder or some other folder is still undecided).

**Comment by BenJKuhn** [comment, marked as answer]
I've opened this as an issue. It's clear that there are quite a few considerations at play, and the right solution is going to require some thoughtful analysis. Please feel free to add notes or make suggestions in the issue thread:

https://github.com/microsoft/WindowsAppSDK/issues/1730
